core/src/main/java/tc/oc/pgm/PGMConfig.java

```diff
@@ -146,6 +155,41 @@
     this.experiments = experiments == null ? ImmutableMap.of() : experiments.getValues(false);
   }
 
+  private static List<GitRepository> parseGitSources(
+      List<String> simpleRepos, List<Map<?, ?>> thoroughRepos) throws URISyntaxException {
+    List<GitRepository> gits = new ArrayList<>();
+
+    for (String repository : simpleRepos) {
+      gits.add(new GitRepository(repository));
+    }
+
+    for (Map<?, ?> repository : thoroughRepos) {
+      try {
+
+        URIish uri = new URIish((String) repository.get("uri"));
+
+        String branch = (String) repository.get("branch");
```

Comment:
Will this default to `master` if not defined?

Reply:
Shouldn't default to `master`, should default to whatever upstream's default is (could be named something else)

Reply:
Yes